extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -107,6 +109,9 @@ MigrationStateBuildItem build(BuildProducer<FeatureBuildItem> featureProducer,
 
         Collection<String> applicationMigrations = applicationMigrationsToDs.values().stream().collect(HashSet::new,
                 AbstractCollection::addAll, HashSet::addAll);
+        for (String applicationMigration : applicationMigrations) {
+            hotDeploymentProducer.produce(new HotDeploymentWatchedFileBuildItem(applicationMigration));
+        }
```

Comment:
Will flyway automatically re-apply changed database migration? IIRC, Flyway versions already applied migration to not re-apply them once the App restarts, I wonder how does this change accounts for that.

Reply:
@machi1990 I believe this all also depends on how the user sets `clean-at-start`, `migrate-at-start`, and also `validate-on-migrate`. Also the user could at any point also use the Dev UI to perform a _Clean_ and / or _Migrate_ action manually.

This probably doesn't answer your question, but I think it will also be the responsibility of the user to know what he is doing. But I think it makes sense that a modified migration script triggers a restart. IMHO the user should then work with `clean-at-start=true` and `migrate-at-start=true`. This is, AFAIK, also what Flyway will set if the migration script is generated from the current Hibernate entities.

Reply:
> This probably doesn't answer your question, but I think it will also be the responsibility of the user to know what he is doing. 

Let's add a NOTE in the documentation for that.

Reply:
Makes sense. I will look into it.

Reply:
Thanks a lot @knutwannheden

Reply:
@machi1990 I've added a tip to the documentation now. I couldn't find a good place for it without doing some minor restructuring. Please let me know if I should revert that again.

Reply:
That's look good. I've noticed that there has not been an addition of a TIP to configure `%dev.quarkus.flyway.clean-at-start=true` in the Flyway doc. Is that intentional?

Reply:
You mean in addition to this? https://github.com/quarkusio/quarkus/pull/25482/files#diff-561acfd06c83e5a429023583b8f55089d355e2c1b370766131bae399f6766eacR143-R148

Reply:
Thanks. I somehow missed that one. It looks good to me.